rust/datafusion/src/optimizer/projection_push_down.rs

```diff
@@ -45,7 +50,7 @@ impl ProjectionPushDown {
     }
 
     fn optimize_plan(
-        &self,
+        &mut self,
```

Comment:
I needed to make this `mut` so that `optimize_explain` can call `OptimizerRule::optimize` which requires a `mut` reference